DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -308,6 +316,112 @@ private DocumentsRunner.CompletionStatus migrateDocumentsWithOneWorker(SourceRep
         }
     }
 
+    public static Stream<Arguments> makeProcessExitArgs() {
+        return Stream.of(
+            Arguments.of(true, 0),
+            Arguments.of(false, 1)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("makeProcessExitArgs")
+    public void testProcessExitsAsExpected(boolean targetAvailable, int expectedExitCode) throws Exception {
+        var sourceImageArgs = makeParamsForBase(SearchClusterContainer.ES_V7_10_2);
+        var baseSourceImageVersion = (SearchClusterContainer.Version) sourceImageArgs[0];
+        var generatorImage = (String) sourceImageArgs[1];
+        var generatorArgs = (String[]) sourceImageArgs[2];
+        var targetImageName = SearchClusterContainer.OS_V2_14_0.getImageName();
+
+        try (var esSourceContainer = new PreloadedSearchClusterContainer(baseSourceImageVersion,
+                    SOURCE_SERVER_ALIAS, generatorImage, generatorArgs);
+                OpensearchContainer<?> osTargetContainer =
+                        new OpensearchContainer<>(targetImageName)) {
+            esSourceContainer.start();
+            osTargetContainer.start();
+
+            final var SNAPSHOT_NAME = "test_snapshot";
+            final List<String> INDEX_ALLOWLIST = List.of();
+            CreateSnapshot.run(
+                    c -> new FileSystemSnapshotCreator(SNAPSHOT_NAME, c, SearchClusterContainer.CLUSTER_SNAPSHOT_DIR),
+                    new OpenSearchClient(esSourceContainer.getUrl(), null),
+                    false);
+            var tempDirSnapshot = Files.createTempDirectory("opensearchMigrationReindexFromSnapshot_test_snapshot");
+            var tempDirLucene = Files.createTempDirectory("opensearchMigrationReindexFromSnapshot_test_lucene");
+
+            String targetAddress = osTargetContainer.getHttpHostAddress();
+
+            String[] args = {
+                "--snapshot-name", SNAPSHOT_NAME,
+                "--snapshot-local-dir", tempDirSnapshot.toString(),
+                "--lucene-dir", tempDirLucene.toString(),
+                "--target-host", targetAddress
+            };
+
+            try {
+                esSourceContainer.copySnapshotData(tempDirSnapshot.toString());
+
+                var targetClient = new OpenSearchClient(targetAddress, null);
+                var sourceRepo = new FileSystemRepo(tempDirSnapshot);
+                migrateMetadata(sourceRepo, targetClient, SNAPSHOT_NAME, INDEX_ALLOWLIST);
+
+                // Stop the target container if we don't want it to be available.  We've already cached the address it was
+                // using, so we can have reasonable confidence that nothing else will be using it and bork our test.
+                if (!targetAvailable) {
+                    osTargetContainer.stop();
+                }
```

Comment:
This is wasteful.  You could bind to a random port (see PortFinder) or setup a local http server if you wanted to keep the connection open.  

Can you document what behavior you're looking the client (migration process) to exhibit?  e.g. "conn_refused means that this part fails, then that part fails, then the process exits"